### PR TITLE
Rearrange turret files to split out functionality

### PR DESCRIPTION
**Pull request by optimumtact**
Add defines for the turret construction phases

Tweak syndicate turret targeting to remove uncessary faction check

Also removes an isrobot check that appears to be an old pre NODROP check for robots adding their laser/laser to the gun.

### CODE REVIEW

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+#define PTURRET_CLOSED = 6
+#define PTURRET_START_EXTERNAL_ARMOUR = 7
+#define PTURRET_EXTERNAL_ARMOUR_ON = 8
+/obj/machinery/porta_turret_construct
+	name = "turret frame"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turret_frame"
+	density=1
+	var/build_step = PTURRET_UNSECURED //the current step in the building process
+	var/finish_name="turret"	//the name applied to the product turret
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: spaces please

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+#define PTURRET_GUN_EQUIPPED = 4
+#define PTURRET_SENSORS_ON = 5
+#define PTURRET_CLOSED = 6
+#define PTURRET_START_EXTERNAL_ARMOUR = 7
+#define PTURRET_EXTERNAL_ARMOUR_ON = 8
+/obj/machinery/porta_turret_construct
+	name = "turret frame"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turret_frame"
+	density=1
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: ditto

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	var/build_step = PTURRET_UNSECURED //the current step in the building process
+	var/finish_name="turret"	//the name applied to the product turret
+	var/installation = null		//the gun type installed
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: aaaa why not just put the actual gun in the turret?

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+					build_step = PTURRET_BOLTED
+					user << "<span class='notice'>You remove the turret's interior metal armor.</span>"
+					new /obj/item/stack/sheet/metal( loc, 2)
+					return
+
+
+		if(PTURRET_INTERNAL_ARMOUR_ON)
+			if(istype(I, /obj/item/weapon/gun/energy)) //the gun installation part
+				var/obj/item/weapon/gun/energy/E = I
+				if(!user.unEquip(I))
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: drop_item

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+					user << "<span class='warning'>\the [I] is stuck to your hand, you cannot put it in \the [src]!</span>"
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: No need for this really

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+				playsound(loc, 'sound/items/Ratchet.ogg', 100, 1)
+				user << "<span class='notice'>You remove the turret's metal armor bolts.</span>"
+				build_step = PTURRET_START_INTERNAL_ARMOUR
+				return
+
+		if(PTURRET_GUN_EQUIPPED)
+			if(isprox(I))
+				build_step = PTURRET_SENSORS_ON
+				if(!user.unEquip(I))
+					user << "<span class='warning'>\the [I] is stuck to your hand, you cannot put it in \the [src]!</span>"
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: nooo no need for these messages... also drop_item

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+			if(istype(I, /obj/item/weapon/weldingtool))
+				var/obj/item/weapon/weldingtool/WT = I
+				if(!WT.isOn()) return
+				if(WT.get_fuel() < 5)
+					user << "<span class='warning'>You need more fuel to complete this task!</span>"
+
+				playsound(loc, pick('sound/items/Welder.ogg', 'sound/items/Welder2.ogg'), 50, 1)
+				user << "<span class='notice'>You begin to weld the turret's armor down...</span>"
+				if(do_after(user, 30/I.toolspeed, target = src))
+					if(!src || !WT.remove_fuel(5, user))
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: Need to check isOn here

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+				user << "<span class='notice'>You pry off the turret's exterior armor.</span>"
+				new /obj/item/stack/sheet/metal(loc, 2)
+				build_step = PTURRET_CLOSED
+				return
+
+	if(istype(I, /obj/item/weapon/pen))	//you can rename turrets like bots!
+		var/t = stripped_input(user, "Enter new turret name", name, finish_name)
+		if(!t)
+			return
+		if(!in_range(src, usr) && loc != usr)
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: `Adjacent`?

File: code/game/machinery/porta_turret/portable_turret_construct.dm [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+		return
+	..()
+
+
+/obj/machinery/porta_turret_construct/attack_hand(mob/user)
+	switch(build_step)
+		if(PTURRET_GUN_EQUIPPED)
+			build_step = PTURRET_INTERNAL_ARMOUR_ON
+
+			var/obj/item/weapon/gun/energy/Gun = new installation(loc)
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_construct.dm]: Yeah no reason not to keep the original gun in contents

File: code/game/machinery/porta_turret/portable_turret_cover.dm [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+************************/
+
+/obj/machinery/porta_turret_cover
+	name = "turret"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	anchored = 1
+	layer = 3.5
+	density = 0
+	var/obj/machinery/porta_turret/Parent_Turret = null
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_cover.dm]: caps are bad

File: code/game/machinery/porta_turret/portable_turret_cover.dm [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+	. = ..()
+	if(.)
+		return
+
+	return Parent_Turret.attack_hand(user)
+
+
+/obj/machinery/porta_turret_cover/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/wrench) && !Parent_Turret.on)
+		if(Parent_Turret.raised) return
```
Comment (neersighted) [inline code/game/machinery/porta_turret/portable_turret_cover.dm]: aaaaaaa why is this seperate

**Comment by optimumtact** [comment]
Thanks for the review but I didn't write that code, it's from the original turret owner

**Comment by optimumtact** [comment]
Addressed as much as I feel I can at this stage.

**Comment by optimumtact** [comment]
I hate the whole concept of tabs

**Comment by optimumtact** [comment]
Merge conflict resolved and adressed the issues @neersighted raised

**Pull request merged**